### DISCUSSION TITLE
Updated release? [0.2.6?] [0.3.0?]

### DISCUSSION THREAD

**Discussion by clintfred** [cross-rs/cross#1290] [General]
Thanks for all your work. Cross is incredibly valuable.

Is there a timeline for an 0.2.6 or 0.3.0 release? We have been depending on the git `main` version to get around an apparent incompatibility when trying to build [zmq-sys](https://github.com/erickt/rust-zmq/tree/master/zmq-sys) with 0.2.5's glibc. This works fine when using `main`s cross, but it would be nice to have a new release to depend on.

**Comment by clintfred** [comment]
I now see 0.3.0 is described [here](https://github.com/cross-rs/cross/milestone/2), so I guess I'm asking for a 0.2.6 release. :smile: 

_edited title for clarity_

**Reply by Emilgardis** [reply]
yes, this is something we really should get to, currently blocked on https://github.com/cross-rs/cross/pull/1208 and I don't know how to fix it myself. I might cut a release without `sparcv9-sun-solaris`.

**Reply by Emilgardis** [reply]
I'm on vacation in august, maybe I can find some motivation and do this myself.

**Reply by clintfred** [reply]
Thanks for the reply. No worries. I was mostly just wondering about the status.

**Comment by Sytten** [comment]
Just wondering how is the 0.3 release going? Anything we can do to help ship it?

**Reply by Emilgardis** [reply]
it's going slowly, I'd want most of [these](https://github.com/cross-rs/cross/milestone/2) to be merged/fixed and then I think we'd be good to go

**Reply by Sytten** [reply]
@Emilgardis just checking on the progress, hopefully can bring a bit of support (moral or otherwise) 🙂
I was wondering if some of the items could be pushed to a 0.3.1. I am guessing most serious people are using the main branch or newer images at the very least these days.
It could be interesting to get some RC out for us to test / report issues more consistently too.
Appreciate all the work 🙏

**Reply by Emilgardis** [reply]
thanks, thats an excellent suggestion, I've moved the non-breaking changes to a new milestone v0.3.1. The few remaining items seem very doable, although I can't promise it will be this week, I think my motivation seeing the new list went up a bit ;)

**Reply by Emilgardis** [reply]
Also ~~not on that list~~ are some breakages in CI, but I think I'll rip the bandaid and just disable those targets until someone really wants it and can fix it. Can make them use the old images that used to work.

edit: I added the breakages as new issues and linked them to the milestone

**Reply by Sytten** [reply]
Excellent! Glad I could help a bit :)
Eagerly waiting for it!!

**Comment by drinkcat** [comment]
Hi @Emilgardis, it's been a year and wondering if 0.3.0 is somewhere on the horizon ,-) -- Or a 0.2.6.

Some context: https://github.com/uutils/coreutils/issues/7392, trying to enable more CI for `uutils/coreutils`. The main problem is that the default docker image is quite old. Of course we can override the image version to `main`, but that means quite a bit of repeated lines in `Cross.toml`, and it'd be nice to have a pinned version as well (instead of a rolling `main` tag that might break at some point?).

Anything I can do to help? Is this list of issues still up to date? https://github.com/cross-rs/cross/milestone/2

**Reply by Emilgardis** [reply]
You can also update cross to the main branch and that would automagically use the images on main.

Im sorry for the delay in getting a new version out, the reason is that my bandwidth is limited and the changes that remain are quite big. Just doing a 0.2.6 would be hard due to the time that has since passed and packages being deprecated.

I wont promise qnything but I feel like I'll be comfortable releasing 0.3.0 soon

**Reply by drinkcat** [reply]
Thanks for the quick reply.

Using cross from main branch isn't really great either for CI, IMHO, we do not really want to be a in a situation were the build breaks because of a change in this repo (it's already questionable for the docker images, but even more for the tool itself...).

**Comment by Sytten** [comment]
Yearly check-in with the project, I feel we are getting close @Emilgardis!

**Reply by nomo-nvm** [reply]
me: wonder why github build works and "newest" tag doesn't
sees: `v0.2.5: 684 commits to main since this release`

Oh...
